### DISCUSSION TITLE
Add recipe to shopping list: Possibility to decide which ingredients should be added

### DISCUSSION THREAD

**Discussion by KidA90** [mealie-recipes/mealie#1876] [Feature Request]
**Before submitting this feature request I have**
- [x] Reviewed the documentation and verified that my feature request isn't already a feature.
- [x] Reviewed the existing feature requests and my feature has not been requested.
- [x] Checked the [tasks taged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

**Please Describe The Problem To Be Solved**
When you add a recipe to the shopping list all ingredients from that recipe are considered. There is no way for the user to decide which ingredients should be added to the shopping list. A possibility to do so would be very helpful if you think of basic ingredients like flour, sugar, salt, pepper etc. There is a high chance that you have these ingredients in storage and don't need to shop them every time they are used in a recipe.

**(Optional): Suggest A Solution**
A really good solution for this would be to have a dialogue box that opens when you add a recipe to the shopping list and lists all the ingredients of the recipe in a check list. The user then could check/uncheck the ingredients and by hitting another "Add to shopping list" button, only the checked items would be added to the shopping list.

Another solution would be something like the features proposed in this thread:
_Originally posted by @Spalfnaganalf1 in https://github.com/hay-kot/mealie/discussions/1733#discussioncomment-4180803_
_"Not sure whether to create a new feature request or add to this one - but i'd like to see the ability to filter the shopping list by the recipe type. The reason for this is that sometimes i have a recipe that includes a sauce, that i may already have frozen or prepared, so before i go shopping, i want to remove those items from the recipe. Would be good to be able to filter based on the recipe so that i know i'm not accidentally removing ingredients from another recipe."_

By filtering or maybe grouping the shopping list by recipe you could also determine which ingredients you need to buy for a recipe and remove the ones you have in storage. Nevertheless I would prefer the first solution, because it would be a bit cumbersome to add a recipe to the shopping list, switch to the shopping list, group by recipe and then delete all the unwanted entries.

**Additional Information**
- [ ] If the feature is accepted I'm willing to submit a PR to provide this feature

**Comment by Orybon** [comment]
I agree with this feature request (i have come recipes with water as ingredient, and i have plenty of water from my tap, no need to buy it from the store :) )

Maybe another solution could be to tick a box or a to put a tag on an ingredient in the data managment page (/group/data/foods). Based on this information some ingredients could be ignored while adding the recipe to the shopping list.

With this solution i think that the exclusion is globally set and no need to set it on each recipe.

**Comment by FuzzyWereBeast** [comment]
am coming over from Paprika. They use a dialogue box as stated earlier. On top of that, they also have a "pantry list" where you keep a running tally of what you have in your pantry. If the item is in the pantry, it is automatically unchecked in the dialogue box.

Because I normally plan for a week, I just add everything to the grocery list. I then "go shopping" in my house before I go shopping at the store. This way I don't accidentally short myself on something.

**Comment by michael-genson** [comment]
FYI this is available now! (has been for a few months): https://github.com/hay-kot/mealie/pull/2136
